### PR TITLE
Revert "[SPARK-40165][BUILD] Update test plugins to latest versions"

### PR DESCRIPTION
**Pull request by Yikun**
### What changes were proposed in this pull request?
This reverts commit 3ed382f391911ec4a79fb204f4986728017dfa4a. https://github.com/apache/spark/pull/37598

### Why are the changes needed?
This patch cause the error like:
```
[error] /home/runner/work/spark/spark/sql/hive-thriftserver/src/main/java/org/apache/hive/service/auth/HttpAuthUtils.java:36:1:  error: package org.apache.http.protocol does not exist
[error] import org.apache.http.protocol.BasicHttpContext;
[error]                                ^
[error] /home/runner/work/spark/spark/sql/hive-thriftserver/src/main/java/org/apache/hive/service/auth/HttpAuthUtils.java:156:1:  error: cannot find symbol
[error]     private final HttpContext httpContext;
[error]                   ^  symbol:   class HttpContext
[error]   location: class HttpKerberosClientAction
[error] 3 errors
```

The old cache + new pom passed. Then, cache is outdate, the real error happened.

According investigation, I found that this https://github.com/apache/spark/commit/3ed382f391911ec4a79fb204f4986728017dfa4a should be the real first bad commit.

I just found this job are using cache:
https://github.com/panbingkun/spark/runs/7938361118?check_suite_focus=true

With old cache + new: passed. (that's the reason that the original PR passed)

Without any cache + new pom: failed. (that's reason that CI break)

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
CI passed, especially, 2.13 passed

### CODE REVIEW

**Comment by Yikun** [comment]
@LuciferYang I'm not sure about this, but if one PR once breaked CI, we'd better revert the whole change to recover CI first and then re-submit new one.

**Comment by Yikun** [comment]
At least 2.13 CI have been passed, let's wait for other CI to be green, then merge this first.

**Comment by LuciferYang** [comment]
Wait a little moment , I only revert `maven-dependency-plugin` in https://github.com/LuciferYang/spark/runs/7955190045?check_suite_focus=true

**Comment by Yikun** [comment]
> And It seems strange that sbt local builds can also succeed :(

You perhaps want to cleanup `org.apache.maven.plugins` and `org.scalacheck` in your local maven repo.

**Comment by LuciferYang** [comment]
> > And It seems strange that sbt local builds can also succeed :(
> 
> You perhaps want to cleanup `org.apache.maven.plugins` and `org.scalacheck` in your local maven repo.

Thanks, let me try this

**Comment by Yikun** [comment]
```yaml
key: build-${{ hashFiles('**/pom.xml', 'project/build.properties', 'build/mvn', 'build/sbt', 'build/sbt-launch-lib.bash', 'build/spark-build-info') }}
restore-keys: |
  build-${{ hashFiles('**/pom.xml', 'project/build.properties', 'build/mvn', 'build/sbt', 'build/sbt-launch-lib.bash', 'build/spark-build-info') }}
```

emmm...BTW, to avoid this kind of problem happened again, I think `restore-keys` should be match strictly to restore cache, otherwise it will use recent `build-` cache, once local repo cache has some dependency error, the CI will not figure out and then give a fake green.

Need some inputs from @HyukjinKwon @dongjoon-hyun WDYT?

**Comment by Yikun** [comment]
It's time to sleep:
- Let's merge this PR to recover CI after this PR green
- We might want to remove `restore-keys` to improve infra to catch similar failure in time.
- After `restore-keys` apply, then re-submit https://github.com/apache/spark/commit/3ed382f391911ec4a79fb204f4986728017dfa4a , make sure cache missed and is not restore, we might see CI failed then fix issue.

**Review by LuciferYang — APPROVED** [review]
+1，LGTM

**Review by HyukjinKwon — APPROVED** [review]

**Review by viirya — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (if CI passes)

**Comment by dongjoon-hyun** [comment]
All tests passed including Scala 2.13. Merged to master.

**Pull request closed without merging**
